### PR TITLE
RLPNC-9054: Changes to record-matching API for multivalued fields

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Review by bsimon-bt — APPROVED** [review]

File: json/src/main/java/com/basistech/rosette/apimodel/jackson/recordsimilaritydeserializers/BooleanFieldDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2025 Basis Technology Corp.
```
Comment (seth-mg) [inline json/src/main/java/com/basistech/rosette/apimodel/jackson/recordsimilaritydeserializers/BooleanFieldDeserializer.java]: The new entity for copyright headers is 'Babel Street Rosette Ltd'

**Review by seth-mg — CHANGES_REQUESTED** [review]
The note on copyrights applies to more than just the one flagged file.

I ran the PR through Sonar and [the results are available here](https://sonar-analytics.babelstreet.com/project/issues?pullRequest=318&issueStatuses=OPEN%2CCONFIRMED&id=com.basistech.rosette%3Arosette-api-java-binding).  There are no Blocker or Critical issues, so you can accept or ignore as you see fit.

Here's a co-pilot review.  The Blocker item in the list is understood, but we'll repeat it for the sake of thoroughness.  Addressing the High and Medium issues seems worthwhile.  If you agree, please address.

**Blocker: public API break in record similarity field types**
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/NameField.java:36,56,66`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/DateField.java:34,54,64`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/AddressField.java:35,54,65`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/StringField.java:27-28`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/NumberField.java:27-28`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/BooleanField.java:27-28`
- usage change example: `examples/src/main/java/com/basistech/rosette/examples/RecordSimilarityExample.java:75-123`

This changes the public construction model from scalar/subtype instances (e.g. `NameField.FieldedName`) to wrapper-with-list (`NameField.builder().data(List.of(...))`). Existing client code targeting `master` will fail to compile and likely break binary compatibility.

**Requested fix:** keep backward-compatible constructors/factories (or adapters) for existing single-value call sites, deprecate old entry points if needed, and layer multivalue support without breaking current consumers.

**High: potential NPE in `toJson()` when data is `null`**
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/NameField.java:43`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/DateField.java:41`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/AddressField.java:42`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/StringField.java:32`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/NumberField.java:32`
- `model/src/main/java/com/basistech/rosette/apimodel/recordsimilarity/records/BooleanField.java:32`

`toJson()` dereferences `data.size()` without a `null` guard. With Lombok builders, `@NotEmpty` does not enforce non-null at build time by itself, so `builder().build()` can produce `data == null` and fail at serialization.

**Requested fix:** enforce non-null in construction (explicit validation in constructor/builder) or fail fast with a clear exception before serialization.

**Medium: missing explicit tests for true multi-element field values**
- `json/src/test/java/com/basistech/rosette/apimodel/RecordSimilarityRequestTest.java:294-313`
- `json/src/test/java/com/basistech/rosette/apimodel/recordsimilarity/RecordSimilarityResponseTest.java:84-89`

Current test updates mostly wrap single values in `List.of(...)` (e.g. one string/one bool/one number). I don’t see coverage for fields containing multiple values in one field payload (the core behavior introduced by this change).

**Requested fix:** add request/response round-trip tests with multi-element `data` arrays (and mixed structured/unstructured entries where allowed), plus negative cases for malformed array elements.

**Comment by ethteck** [comment]
@seth-mg I think I've addressed everything necessary, but let me know if I missed something

**Review by seth-mg — APPROVED** [review]
LGTM!  (I thought I approved it yesterday)
